src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,187 @@
+    passthrough_args: Optional[str] = None,
+    interpreter_constraints: Optional[str] = None,
+    skip: bool = False,
+  ) -> LintResult:
+    args = ["--backend-packages2=pants.backend.python.lint.pylint"]
+    if config:
+      # TODO: figure out how to get this file to exist...
+      self.create_file(relpath="pylintrc", contents=config)
+      args.append("--pylint-config=pylintrc")
+    if passthrough_args:
+      args.append(f"--pylint-args='{passthrough_args}'")
+    if skip:
+      args.append(f"--pylint-skip")
+    input_snapshot = self.request_single_product(Snapshot, InputFilesContent(source_files))
+    target = PylintTarget(
+      PythonTargetAdaptor(
+        sources=EagerFilesetWithSpec(self.source_root, {'globs': []}, snapshot=input_snapshot),
+        address=Address.parse(f"{self.source_root}:target"),
+        compatibility=[interpreter_constraints] if interpreter_constraints else None,
+      )
+    )
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.good_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source])
+    assert result.exit_code == 16
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source, self.bad_source])
+    assert result.exit_code == 16
+    assert "good.py" not in result.stdout
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  @pytest.mark.skip(reason="Get config file creation to work with options parsing")
+  def test_respects_config_file(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], config="[pylint]\ndisable = C0114\n")
+    assert result.exit_code == 0
+    assert result.stdout.strip() == ""
+
+  def test_respects_passthrough_args(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], passthrough_args="--disable=C0114")
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_transitive_dep(self) -> None:
```

Comment:
Consider renaming to something like `test_includes_direct_dependencies` and/or adding a comment explaining how we need direct dependencies but don't care about transitive.